### PR TITLE
fixes #185: Consumer not working in Cluster environment

### LINKED ISSUE #185: Consumer not working in Cluster environment
Consumer not working in Cluster environment because of they are all in FOLLOWER state and the `Neo4jUtils.isWriteableInstance` blocks the write.
We have to wait that the RAFT election ends

### PR DESCRIPTION
**Pull request by conker84**
Fixes #185 

## Proposed Changes (Mandatory)

A brief list of proposed changes in order to fix the issue:

  - Wait for the leader election in case of a cluster environment

### CODE REVIEW

File: common/src/main/kotlin/streams/utils/Neo4jUtils.kt [hunk 1]
```diff
@@ -32,4 +40,61 @@ object Neo4jUtils {
...
+            action()
+        }
+    }
+
+    fun executeInLeader(db: GraphDatabaseAPI, log: Log, timeout: Long = 50000, action: () -> Unit) {
+        if (isEnterpriseEdition(db)) {
+            // wait for the election
+            GlobalScope.launch(Dispatchers.IO) {
+                val start = System.currentTimeMillis()
+                var end = System.currentTimeMillis()
```
Comment (jexp) [inline common/src/main/kotlin/streams/utils/Neo4jUtils.kt]: we don't have to store `end`

File: consumer/src/main/kotlin/streams/StreamsEventSinkExtensionFactory.kt [hunk 2]
```diff
@@ -51,27 +51,21 @@ class StreamsEventSinkExtensionFactory : KernelExtensionFactory<StreamsEventSink
...
+                                        log,
+                                        db)
+                        // start the Sink
+                        Neo4jUtils.executeInLeader(db, log) { initSinkModule() }
 
                         // Register required services for the Procedures
                         StreamsSinkProcedures.registerStreamsSinkConfiguration(streamsSinkConfiguration)
                         StreamsSinkProcedures.registerStreamsEventConsumerFactory(eventSink.getEventConsumerFactory())
                         StreamsSinkProcedures.registerStreamsEventSinkConfigMapper(eventSink.getEventSinkConfigMapper())
-                        streamsLog.info("Streams Sink module initialised")
```
Comment (jexp) [inline consumer/src/main/kotlin/streams/StreamsEventSinkExtensionFactory.kt]: why did you remove this?

Comment (conker84) [reply consumer/src/main/kotlin/streams/StreamsEventSinkExtensionFactory.kt]: I moved it into the `initSinkModule` method

File: common/src/main/kotlin/streams/utils/Neo4jUtils.kt [hunk 3]
```diff
@@ -32,4 +40,61 @@ object Neo4jUtils {
         return db.dependencyResolver
                 .resolveDependency(LogService::class.java)
     }
+
+    fun isEnterpriseEdition(db: GraphDatabaseAPI): Boolean {
+        try {
+            db.execute("CALL dbms.cluster.role()").columnAs<String>("role").next()
```
Comment (mneedham) [inline common/src/main/kotlin/streams/utils/Neo4jUtils.kt]: For checking for version, we can use this procedure:

```
CALL dbms.components() YIELD edition
RETURN edition = "enterprise" AS isEnterprise
```

Comment (conker84) [reply common/src/main/kotlin/streams/utils/Neo4jUtils.kt]: Thanks @mneedham I changed the method with your suggestion!

**Pull request merged**